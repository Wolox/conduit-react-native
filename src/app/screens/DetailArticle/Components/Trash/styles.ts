import { StyleSheet } from 'react-native';
import { red } from '@constants/colors';

const COMMON_SIZE_TRASH = 18;

const styles = StyleSheet.create({
  containerTrash: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 5,
    width: '8%'
  },
  icTrash: {
    height: COMMON_SIZE_TRASH,
    width: COMMON_SIZE_TRASH,
    tintColor: red
  }
});

export default styles;
