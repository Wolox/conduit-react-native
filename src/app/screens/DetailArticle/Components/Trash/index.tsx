import React from 'react';
import { TouchableOpacity, Image } from 'react-native';
import icTrash from '@assets/icons/icTrash.png';

import styles from './styles';

interface Props {
  username?: string;
  currentUser?: string;
  onPress?: () => void;
}
export default function Trash({ username, currentUser, onPress }: Props) {
  return (
    <>
      {username && currentUser === username && (
        <TouchableOpacity style={styles.containerTrash} onPress={onPress}>
          <Image source={icTrash} resizeMode="contain" resizeMethod="scale" style={styles.icTrash} />
        </TouchableOpacity>
      )}
    </>
  );
}
