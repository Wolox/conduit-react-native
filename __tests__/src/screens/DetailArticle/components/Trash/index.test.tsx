import React from 'react';
import renderer from 'react-test-renderer';
import Trash from '@screens/DetailArticle/components/Trash';

describe('Trash', () => {
  test('Trash Snapshot', () => {
    const tree = renderer.create(<Trash />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
