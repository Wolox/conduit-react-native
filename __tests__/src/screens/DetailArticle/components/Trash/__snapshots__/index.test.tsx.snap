// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trash Trash Snapshot 1`] = `null`;
